### PR TITLE
contributing: add DCO section & small changes

### PR DESCRIPTION
**Pull request by unclejack**
This PR brings some of the changes made to the same file on the netplugin repository.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,22 +1,122 @@
...
+Please make sure to run the tests and that the tests pass before
+submitting the PR. Please keep in mind that some changes might not be
+merged if the maintainers decide they can't be merged.
+
+Please squash your commits to one commit per fix or feature. The resulting
+commit should have a single meaningful message.
+
+## Commit message guidelines
+
+The commit should have a short summary of no more than 50 characters on
```
Comment (erikh) [inline CONTRIBUTING.md]: let's leave the line limits out -- I don't really feel this is a necessary restriction, and I break this rule pretty much every commit.

**Comment by unclejack** [comment]
@erikh: PTAL

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,22 +1,120 @@
...
+fixes, propose design changes, discuss use cases or become a maintainer.
 
-* Sign your commits with `git commit -s`. Note that we live under an Apache
-  license.
-* Your comments should detail the directory they are affecting, as below. This
-  makes it much simpler than `git log --stat` to determine what areas of a
-  project a specific commit is referencing.
-  * Example: `librbd: changed a widget`
-* Please squash your commits per fix/feature, with a single commit message.
+## Submitting pull requests to change the documentation or the code
```
Comment (erikh) [inline CONTRIBUTING.md]: gah, forgot this, sorry. Please leave the restriction in about directory prefixing the commit messages.

Comment (unclejack) [inline CONTRIBUTING.md]: This is now below in the section about the commit message guidelines.
```
+The short summary should include the name of the directory affected by
+the commit (e.g.: `librbd: change a widget`).
```

Comment (erikh) [inline CONTRIBUTING.md]: seems good! let's merge this.

**Pull request merged**